### PR TITLE
Band-aid for Bulk SDKHooks performance impact

### PR DESCRIPTION
**Pull request by KaelaSavia**
Resolves https://github.com/alliedmodders/sourcemod/issues/1935 

Original solution found by https://github.com/alliedmodders/sourcemod/issues/1935#issuecomment-1513620472

### CODE REVIEW

**Comment by dvander** [comment]
Just remove the code or make it platform dependent.(Not commenting on the validity of the change but commented code is a style/readability problem)On Dec 6, 2023 10:36 AM, Kaela ***@***.***> wrote:Resolves #1935
Original solution found by #1935 (comment)



You can view, comment on, or merge this pull request online at:
  https://github.com/alliedmodders/sourcemod/pull/2092

Commit Summary
cf48f65  Band-aid for Bulk SDKHooks performance impact

File Changes  (1 file)

    M
    extensions/sdkhooks/extension.cpp
    (6)
  

Patch Links:
https://github.com/alliedmodders/sourcemod/pull/2092.patchhttps://github.com/alliedmodders/sourcemod/pull/2092.diff

**Comment by KaelaSavia** [comment]
Gotcha.

As for making code platform dependent, I believe it would complicate things unnecessarily and would serve no functional benefit in this case!

**Comment by Headline** [comment]
> Gotcha.
> 
> As for making code platform dependent, I believe it would complicate things unnecessarily and would serve no functional benefit in this case!

If the issue is limited to one platform, and this is a "bandaid" fix, scope should probably be limited.

KyleS mentioned us resetting the table on levelend, seems that's missing here

**Comment by KaelaSavia** [comment]
I'll make proper pull request at later date probably as I'll be busy during holidays.

Fixes seem to work however including resetting of hooks on map change so if anyone feels like continuing work then sure!

File: extensions/sdkhooks/extension.cpp [hunk 1]
```diff
@@ -445,6 +450,33 @@ void SDKHooks::OnClientDisconnecting(int client)
...
 
+void SDKHooks::LevelShutdown()
+{
+#if defined PLATFORM_LINUX
+	for (size_t type = 0; type < SDKHook_MAXHOOKS; ++type)
+	{
+		std::vector<CVTableList *> &vtablehooklist = g_HookList[type];
+		for (size_t listentry = 0; listentry < vtablehooklist.size(); ++listentry)
+		{
+			std::vector<HookList> &pawnhooks = vtablehooklist[listentry]->hooks;
```
Comment (KyleSanderson) [inline extensions/sdkhooks/extension.cpp:461]: there should be a zap function or something to just clear the vector.

File: extensions/sdkhooks/extension.cpp [hunk 2]
```diff
@@ -445,6 +450,33 @@ void SDKHooks::OnClientDisconnecting(int client)
...
+			for (size_t entry = 0; entry < pawnhooks.size(); ++entry)
+			{
+				pawnhooks.erase(pawnhooks.begin() + entry);
+				entry--;
+			}
+
+			if (pawnhooks.size() == 0)
```
Comment (KyleSanderson) [inline extensions/sdkhooks/extension.cpp:468]: not needed.

File: extensions/sdkhooks/extension.cpp [hunk 3]
```diff
@@ -445,6 +450,33 @@ void SDKHooks::OnClientDisconnecting(int client)
...
+			{
+				delete vtablehooklist[listentry];
+				vtablehooklist.erase(vtablehooklist.begin() + listentry);
```
Comment (KyleSanderson) [inline extensions/sdkhooks/extension.cpp:471]: ditto on zap / clear, can happen at the end outside of the loop.

**Pull request closed without merging**